File: Difficulty: Medium/Longest substring with distinct characters/longest-substring-with-distinct-characters.cpp
class Solution {
  public:
    int longestUniqueSubstr(string &s) {
        // code here
        int n = s.size();
        
        vector<int> check(26,-1);
        
        int i = 0, j = 0;
        
        
        int ans = 0;
        
        while(j<n) {
            
            if(check[s[j]-'a']!=-1) {
                
                i = max(i,check[s[j]-'a']+1);
                
            }
            
            check[s[j]-'a'] = j;
            
            ans = max(ans,j-i+1);
            j++;
            
        }
        
        return ans;
    }
};

File: Difficulty: Medium/Longest Substring with Distinct Characters/longest-substring-with-distinct-characters.java
class Solution {
    public int longestUniqueSubstr(String s) {
        // code here
        
        int n = s.length();
        int[] charArr = new int[26];
        for(int i = 0;i<26;i++) charArr[i] = -1;
        
        int j = 0;
        int i = 0;
        int ans = 1;
        
        while(j<n) {
            
            
            if(charArr[s.charAt(j)-'a']!=-1) {
                i = Math.max(i,charArr[s.charAt(j)-'a']+1);
            }
            
            charArr[s.charAt(j)-'a'] = j;
            
            ans = Math.max(ans,j-i+1);
            
            j++;
        }
        return ans;
    }
}